chat/consumers.py

```diff
@@ -0,0 +1,43 @@
+import json
+from channels.generic.websocket import AsyncWebsocketConsumer
+
+class ChatConsumer(AsyncWebsocketConsumer):
+    """
+    WebSocket consumer for handling real-time chat messages in specific rooms.
+    """
+
+    async def connect(self):
+        """
+        Handles a new WebSocket connection.
+        Joins the corresponding room group based on the room name.
+        """
+        self.room_name = self.scope["url_route"]["kwargs"]["room_name"]
```

Comment:
If room_name can contain unsafe or untrusted input (e.g., from a URL), sanitize or validate it before using it as a group name to prevent injection or unexpected group joining.